### PR TITLE
docs: convert to CSF Stories with MDX docs

### PR DESCRIPTION
**Pull request by nineteen88**
### Proposed behaviour

Rewrite any stories for the remaining Typescript components that haven't been changed to the CSF story approach.

### Current behaviour

Currently there are a number of Typescript components that were refactored before we started using the CSF story approach.

### Checklist

- [X] Commits follow our style guide
- [X] Unit tests added or updated if required
- [X] Storybook added or updated if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/box/box.stories.tsx [hunk 1]
```diff
@@ -217,6 +217,7 @@ export const OverflowWrap: ComponentStory<typeof Box> = () => {
     </div>
   );
 };
+OverflowWrap.parameters = { chromatic: { disable: true } };
```
Comment (DlgSHi) [inline src/components/box/box.stories.tsx:220]:
```suggestion
OverflowWrap.parameters = { chromatic: { disableSnapshot: true } };
```

File: src/components/box/box.stories.tsx [hunk 2]
```diff
@@ -285,3 +286,4 @@ export const Scroll: ComponentStory<typeof Box> = () => {
     </div>
   );
 };
+Scroll.parameters = { chromatic: { disable: true } };
```
Comment (DlgSHi) [inline src/components/box/box.stories.tsx:289]: and for other as well

```suggestion
Scroll.parameters = { chromatic: { disableSnapshot: true } };
```

File: src/components/textbox/textbox.stories.mdx [hunk 3]
```diff
@@ -3,9 +3,11 @@ import { Meta, Story, Canvas } from "@storybook/addon-docs";
 
 import StyledSystemProps from "../../../.storybook/utils/styled-system-props";
 import TranslationKeysTable from "../../../.storybook/utils/translation-keys-table";
+import CarbonProvider from "../carbon-provider/carbon-provider.component";
```
Comment (Parsium) [inline src/components/textbox/textbox.stories.mdx:6]:
```suggestion
```

**Review by edleeks87 — COMMENTED** [review]
Nothing from me that hasn't already been raised

File: src/components/menu/index.d.ts [hunk 4]
```diff
@@ -3,3 +3,4 @@ export { default as MenuItem } from "./menu-item/menu-item";
 export { default as ScrollableBlock } from "./scrollable-block/scrollable-block";
 export { default as MenuDivider } from "./menu-divider/menu-divider";
 export { default as MenuSegmentTitle } from "./menu-segment-title/menu-segment-title";
+export { default as MenuFullscreen } from "./menu-full-screen/menu-full-screen";
```
[the remark this answers is not in the document]
Comment (nineteen88) [reply src/components/menu/index.d.ts:6]: I had to because of typescript errors since it couldn't find MenuFullscreen in one of the components

**Review by Parsium — DISMISSED** [review]

File: src/components/button-bar/button-bar.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    <ButtonBar size="large" ml={2} mt={2}>
+      <Button subtext="subtext 1">Large</Button>
+      <Button subtext="subtext 2">Large</Button>
+      <Button subtext="subtext 3">Large</Button>
+    </ButtonBar>
+  </>
+);
+
+export const buttonBarIcons: ComponentStory<typeof ButtonBar> = () => {
+  const BUTTON_BAR_SIZES = ["small", "medium", "large"] as const;
```
[the remark this answers is not in the document]
Comment (nineteen88) [reply src/components/button-bar/button-bar.stories.tsx:35]: It's to add context to the stories. If we pull the values from the config then all you would see in the code example is a variable name. Defining the variables within the story allows the end user to understand the implementation detail in full.

File: src/components/textbox/textbox.stories.mdx
[the remark this answers is not in the document]
Comment (nineteen88) [reply src/components/textbox/textbox.stories.mdx:1]: Strange one, it's seeing a comment that contained `<id>` as a component and trying to render it. Changed to `[id]` instead.

File: src/components/button-toggle/button-toggle.stories.mdx
[the remark this answers is not in the document]
Comment (nineteen88) [reply src/components/button-toggle/button-toggle.stories.mdx:1]: Believe this was fixed as part of a change following James' feedback. I can't see the error message, but if it's still there let me know

**Review by Parsium — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Pull request merged**